### PR TITLE
Add `show reboot-cause history` cli

### PR DESCRIPTION
**Pull request by sujinmkang**
- What I did
To make the reboot cause history information available for telemetry service, during processing the reboot-cause on boot, the information is saved in state-db.
The cli command show reboot-cause is changed to support the new file format and file name.
New command show reboot-cause history is added to display the reboot-cause information which is stored in state-db.
- How I did it
Add the 'show reboot-cause history' command to read up to the10 previous reboot-cause information from state-db and display them.
Read the last reboot-cause from the new file "previous-reboot-cause.json" and process the data to display for show reboot-cause

- How to verify it
Unit test added.
Verified also in DUT

- Previous command output (if the output of a command-line utility has changed)

- New command output (if the output of a command-line utility has changed)
```bash
admin@sonic:~$ show reboot-cause history
name                 cause        time                          user    comment
-------------------  -----------  ----------------------------  ------  ---------
2020_10_09_04_53_58  warm-reboot  Fri Oct  9 04:51:47 UTC 2020  admin
2020_10_09_02_33_06  reboot       Fri Oct  9 02:29:44 UTC 2020  admin
2020_10_09_02_00_53  fast-reboot  Fri Oct  9 01:58:04 UTC 2020  admin
2020_10_09_01_56_59  reboot       Fri Oct  9 01:53:49 UTC 2020  admin
```
HLD : Azure/SONiC#669

### CODE REVIEW

File: show/main.py [hunk 1]
```diff
@@ -1813,23 +1816,55 @@ def mmu():
...
 
 #
-# 'reboot-cause' command ("show reboot-cause")
+# 'reboot-cause' group ("show reboot-cause")
 #
-@cli.command('reboot-cause')
-def reboot_cause():
-    """Show cause of most recent reboot"""
-    PREVIOUS_REBOOT_CAUSE_FILE = "/host/reboot-cause/previous-reboot-cause.txt"
+def readRebootCauseFile():
```
Comment (jleveque) [inline show/main.py:1821]: Rename `read_reboot_cause_file()``

**Review by jleveque — CHANGES_REQUESTED** [review]
As comments. Also, since `reboot-cause` is now a group, can you please move this functionality to a new file, `reboot_cause.py` and import the file in main.py? There are a number of other examples.

File: show/reboot_cause.py [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+PREVIOUS_REBOOT_CAUSE_FILE = "/host/reboot-cause/previous-reboot-cause.json"
+USER_ISSUED_REBOOT_CAUSE_REGEX ="User issued \'{}\' command [User: {}, Time: {}]"
+
+
+#
+# 'reboot-cause' group ("show reboot-cause")
+#
+def read_reboot_cause_file():
+    result=""
```
Comment (jleveque) [inline show/reboot_cause.py:17]: Add spaces around `=`

File: show/reboot_cause.py [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+import click
+
+PREVIOUS_REBOOT_CAUSE_FILE = "/host/reboot-cause/previous-reboot-cause.json"
+USER_ISSUED_REBOOT_CAUSE_REGEX ="User issued \'{}\' command [User: {}, Time: {}]"
+
+
+#
+# 'reboot-cause' group ("show reboot-cause")
+#
```
Comment (jleveque) [inline show/reboot_cause.py:15]: This comment doesn't apply to this function

File: show/reboot_cause.py [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+        with open(PREVIOUS_REBOOT_CAUSE_FILE) as f:
+            result = json.load(f)
+    return result
+
+@click.group(name='reboot-cause', short_help='Show cause of most recent reboot', cls=clicommon.AliasedGroup, invoke_without_command=True)
+@click.pass_context
+def reboot_cause(ctx):
+    if ctx.invoked_subcommand is None:
+        last_reboot_cause = ""
+        # Read the last previous reboot cause
```
Comment (jleveque) [inline show/reboot_cause.py:28]: Remove the work "last"

**Review by jleveque — CHANGES_REQUESTED** [review]

**Comment by lguohan** [comment]
please paste code coverage number.

**Comment by sujinmkang** [comment]
retest this please!

File: show/reboot_cause.py [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+    REBOOT_CAUSE_TABLE_NAME = "REBOOT_CAUSE"
+    TABLE_NAME_SEPARATOR = '|'
+    db = SonicV2Connector(host='127.0.0.1')
+    db.connect(db.STATE_DB, False)   # Make one attempt only
+    prefix = REBOOT_CAUSE_TABLE_NAME + TABLE_NAME_SEPARATOR
+    _hash = '{}{}'.format(prefix, '*')
+    table_keys = db.keys(db.STATE_DB, _hash)
+    if table_keys is None:
+        click.echo("Reboot-cause history is not yet available in StateDB")
+        exit()
```
Comment (jleveque) [inline show/reboot_cause.py:51]: Prefer `sys.exit()` to `exit()` (will also need to import sys)

**Review by jleveque — CHANGES_REQUESTED** [review]

**Comment by sujinmkang** [comment]
retest this please!

File: show/reboot_cause.py (hunk 4 above)
Comment (jleveque) [reply show/reboot_cause.py:30]: Still not resolved

File: tests/reboot_cause_test.py [hunk 6]
```diff
@@ -0,0 +1,68 @@
+import os
+import sys
+import textwrap
+
+import mock
+
```
Comment (jleveque) [inline tests/reboot_cause_test.py:6]: Remove this blank line. `mock` and `click` are both thrid-party packages and should be in the same group.

**Review by jleveque — CHANGES_REQUESTED** [review]

File: show/reboot_cause.py (hunk 4 above)
Comment (sujinmkang) [reply show/reboot_cause.py:30]: did you mean the comment? :) ok!

**Comment by sujinmkang** [comment]
retest this please

File: show/reboot_cause.py [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+        r.append(tk.replace(prefix,""))
+        r.append(entry['cause'] if 'cause' in entry else "")
+        r.append(entry['time'] if 'time' in entry else "")
+        r.append(entry['user'] if 'user' in entry else "")
+        r.append(entry['comment'] if 'comment' in entry else "")
+        table.append(r)
+
+    header = ['Name', 'Cause', 'Time', 'User', 'Comment']
+    click.echo(tabulate(table, header))
+
```
Comment (jleveque) [inline show/reboot_cause.py:69]: Remove blank line from EOF

**Review by jleveque — CHANGES_REQUESTED** [review]

**Review by jleveque — APPROVED** [review]

**Comment by sujinmkang** [comment]
retest this please

**Comment by sujinmkang** [comment]
retest this please

**Comment by lguohan** [comment]
can you fix unit test failure?

**Comment by sujinmkang** [comment]
@lguohan yes, I'm checking it. I got some other errors on sonic-utilities builds so I'm doing a clean up build to reproduce this unit test failure.

File: show/main.py [hunk 8]
```diff
@@ -16,6 +16,7 @@
 from tabulate import tabulate
 from utilities_common.db import Db
 
+from . import reboot_cause
```
Comment (jleveque) [inline show/main.py:19]: Please insert this line in alphabetical order

File: show/main.py [hunk 9]
```diff
@@ -126,6 +127,7 @@ def cli(ctx):
 
 
 # Add groups from other modules
+cli.add_command(reboot_cause.reboot_cause)
```
Comment (jleveque) [inline show/main.py:130]: Please insert this line in alphabetical order

**Review by jleveque — CHANGES_REQUESTED** [review]

**Comment by sujinmkang** [comment]
[2020-11-28T07:13:01.701Z] ----------- coverage: platform linux, python 3.7.3-final-0 -----------
[2020-11-28T07:13:01.701Z] Name                                          Stmts   Miss  Cover
[2020-11-28T07:13:01.701Z] -----------------------------------------------------------------
[2020-11-28T07:13:01.702Z] show/reboot_cause.py                             48      4    92%

**Review by jleveque — APPROVED** [review]

**Pull request merged**